sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ConflictResolutionPolicy.java

```diff
@@ -128,6 +132,54 @@ public static ConflictResolutionPolicy createCustomPolicy(String conflictResolut
         return policy;
     }
 
+    /**
+     * Creates a CUSTOM {@link ConflictResolutionPolicy} which uses the specified stored procedure
+     * to perform conflict resolution
+     * <p>
+     * This stored procedure may be created after the {@link CosmosContainerProperties} is created and can be changed as
+     * required.
+     *
+     * <ul>
+     * <li>In case the stored procedure fails or throws an exception,
+     * the conflict resolution will default to registering conflicts in the conflicts feed</li>
+     * </ul>
+     *
+     * @param storedProcedure stored procedure to perform conflict resolution.
+     * @return ConflictResolutionPolicy.
+     */
+    public static ConflictResolutionPolicy createCustomPolicy(CosmosStoredProcedure storedProcedure) {
```

Comment:
Adding this api because as of now customer need to pass the self link by themselves, and in V4 we don't have a way of getting self link from resources. Only way for user is if they manually create the URL.

Reply:
removed this api